repositories/tokenRepository.go

```diff
@@ -0,0 +1,32 @@
+package repositories
+
+import (
+	"context"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/x/bsonx"
+)
+
+// TokenRepository is the repository for Service Token objects
+type TokenRepository struct {
+	*mongo.Collection
+}
+
+const tokenCollection = "tokens"
+
+// NewTokenRepository creates a new TokenRepository
+func NewTokenRepository(db *mongo.Database) (*TokenRepository, error) {
+	_, err := db.Collection("tokens").Indexes().CreateOne(
```

Comment:
I decided to include the creator ID since you would only want to display service tokens to the user that created them, although there probably won't be too many entries, I thought it was worthwhile to include it - especially given how we'd need to query the DB to get the data for the frontend

Reply:
Strictly speaking isn't it's necessary to have an index for the creator though